Notes-Azure.adoc

```diff
@@ -0,0 +1,147 @@
+
+=== Requirements
+
+To setup the Microsoft Azure environment the command line client *az* is required.
+
+* On SUSE systems it can be installed by executing the following commands:
++
+[source,bash]
+----
+$ sudo zypper install -y curl
+$ sudo rpm --import https://packages.microsoft.com/keys/microsoft.asc
+$ sudo sh -c 'echo -e "[azure-cli]\nname=Azure CLI\nbaseurl=https://packages.microsoft.com/yumrepos/azure-cli\nenabled=1\ntype=rpm-md\ngpgcheck=1\ngpgkey=https://packages.microsoft.com/keys/microsoft.asc" > /etc/zypp/repos.d/azure-cli.repo'
+$ sudo zypper refresh
+$ sudo zypper install -y azure-cli
+----
+
+* Login to Azure
++
+[source,bash]
+----
+$ az login
+----
+
+=== Setup of Kubernetes
+
+* The subscriptions ID needs to be exctracted and written down:
++
+[source,bash]
+----
+$ az account show --query "{ subscription_id: id }"
+----
+
+
+* Create a service principal and write down the _password_ and _appId_.
+The term _<subscription-id>_ needs to be replaced by the _subscription_id_ noted in the in the previous step. If the Microsoft Azure service is supposed to be located in another data center then in the Western USA _westus_ needs to be replaced. 
++
+[source,bash]
+----
+$ export SUBSCRIPTION_ID=<subscription-id>
+$ az account set --subscription $SUBSCRIPTION_ID
+$ az group create --name scf-resource-group --location westus
+$ az ad sp create-for-rbac --role Contributor --scopes "/subscriptions/$SUBSCRIPTION_ID/resourceGroups/scf-resource-group"
+----
+
+* Create a container service in Azure and replace the terms _<password>_ and _<application-id>_ by the one noted from the previous step. Additionally the phrase _<your-publi- ssh-key>_ should be replaced by the public ssh key to be used:
++
+[source,bash]
+----
+$ az acs create \
+       --name scf-container-service \
+       --resource-group scf-resource-group \
+       --orchestrator-type Kubernetes \
+       --dns-prefix "myscf" \
+       --master-count 1 \
+       --admin-username scf-admin \
+       --agent-count 3 \
+       --client-secret <password> \
+       --ssh-key-value "<your-publi- ssh-key>" \
+       --service-principal <application-id> \
+       --master-vm-size Standard_D2_v2
+----
+
+* Prepare kubernetes environment. If there is already another running Kubernetes it is advised to backup the old configuration located in '~/.kube/'.
++
+[source,bash]
+----
+$ az acs kubernetes get-credentials --resource-group="scf-resource-group" --name="scf-container-service"
+----
+
+* The Kubernetes config can be verified by listing all the current pods.
++
+[source,bash]
+----
+$ kubectl get pods --all-namespaces
```

Comment:
When I run this command for the first time, it failed with the error:

`error: group map[foobar] is already registered`

After a quick search I found out that a newer version of kubernetes is needed. I was using the kubernetes-client package version 1.3.10-1.15 from openSUSE Leap 42.3 which is apparently too old. After updating to version 1.6.1 from the Virtualization:containers repo the call was successful.

I don't know which version at least is needed for this command to succeed, but we need to add a note here making people aware that they might need a newer kubernetes-client.